src/protobuf_impl.rs

```diff
@@ -9,45 +9,41 @@ use std::path::Path;
 use std::process::Command;
 use std::str::from_utf8;
 
-// For preference we use the protoc from bin which we bundle with the crate. If
-// there is not one suitable for the platform, then we try system protoc.
+// We use system protoc when its version matches,
+// otherwise use the protoc from bin which we bundle with the crate.
 fn get_protoc() -> String {
+    let ver_re = Regex::new(r"([0-9]+)\.([0-9]+)\.[0-9]").unwrap();
```

Comment:
Rather than inline this code, could `check_protoc_version` return an `Option` and `generate_files` can skip calling `get_protoc` if the system protoc is found?

Reply:
Previously, the `generate_files` always check the protoc version. I'd rather check the version and fall back to the bundled protoc when the version mismatches.
Another option is to use the `PROTOC` environment variable as prost do (https://github.com/danburkert/prost/blob/master/prost-build/build.rs#L4-L8)

Reply:
The substantial change is fine, I just prefer to keep a `check_protoc_version` function separate to make the code easier to read. Is it ok to do?